iron-location.html

```diff
@@ -360,18 +364,27 @@
         var origin;
 
         // IE Polyfill
-        if (window.location.origin) {
-          origin = window.location.origin;
+        if (this.__location.origin) {
+          origin = this.__location.origin;
         } else {
-          origin = window.location.protocol + '//' + window.location.host;
+          origin = this.__location.protocol + '//' + this.__location.host;
         }
 
         var urlOrigin;
 
         if (url.origin) {
           urlOrigin = url.origin;
         } else {
-          urlOrigin = url.protocol + '//' + url.host;
+          // IE always adds port number on HTTP and HTTPS on <a>.host but not on
+          // window.location.host
+          var protocolRegex = /:(80|443)$/;
+          var urlHost = url.host;
```

Comment:
I think it would be easier and clearer to access `url.port` and `url.hostname` rather than doing regex parsing of `url.host`

Reply:
Related, it'd be good to only elide the port if it matches the protocol

Reply:
im dumb, this is a better idea